examples/python/ibmq/using_qiskit_terra_level_2.py

```diff
@@ -46,60 +37,52 @@
 
 # Making another circuit: GHZ State
 qc2 = QuantumCircuit(4, 4)
-qc2.h([0, 1, 2, 3])
+qc2.h([0])
 qc2.cx(0, 1)
 qc2.cx(0, 2)
 qc2.cx(0, 3)
 qc2.measure([0, 1, 2, 3], [0, 1, 2, 3])
 
+
+def status_header(sort):
+    print(f"{sort:<25} |  jobs | status")
+    print("===========================================")
+
+
+def format_status(status):
+    status = status.to_dict()
+    print(f"{status['backend_name'][:25]:>25} | {status['pending_jobs']:5d} | {status['status_msg'][:6]:>6}")
+
+
 # Setting up the backend
-print("(Aer Backends)")
+status_header("Aer Backends")
 for backend in BasicAer.backends():
-    print(backend.status())
+    format_status(backend.status())
 qasm_simulator = BasicAer.get_backend('qasm_simulator')
 
 
 # Compile and run the circuit on a real device backend
 # See a list of available remote backends
-print("\n(IBMQ Backends)")
+status_header("IBMQ Backends")
 for backend in provider.backends():
-    print(backend.status())
+    format_status(backend.status())
 
 try:
     # select least busy available device and execute.
-    least_busy_device = least_busy(provider.backends(simulator=False))
+    least_busy_device = least_busy([backend for backend in provider.backends(simulator=False)
+                                    if backend.status().pending_jobs > 0])
 except:
     print("All devices are currently unavailable.")
 
 print("Running on current least busy device: ", least_busy_device)
 
 
-# making a pass manager to compile the circuits
+# compiling the circuits
 coupling_map = CouplingMap(least_busy_device.configuration().coupling_map)
 print("coupling map: ", coupling_map)
 
-pm = PassManager()
-
-# Use the trivial layout
-pm.append(TrivialLayout(coupling_map))
-
-# Extend the the dag/layout with ancillas using the full coupling map
-pm.append(FullAncillaAllocation(coupling_map))
-pm.append(EnlargeWithAncilla())
```

Comment:
The missing step here I believe is `pm.append(ApplyLayout())` after `EnlargeWithAncilla()`. The argument to the `Unroller` would also likely need to be updated to `leasy_busy_device.configuration().basis_gates`.

Reply:
Modern `Unroller` seems to have bad behavior here: it unrolls `H` into a `U2`, then a `U3`, then a `U`, then vomits that it doesn't know how to unroll `U` any further. Perhaps once upon a time the definition of `H` was in terms of `RZ` and `SX` and this would work — but not any longer, it seems.

I replaced this with a synthesis pass instead. Might be overkill.

Reply:
Nah, this is how the unroller always worked. It is a very basic algorithm it just descends through the definition attribute of gates and hopes it converges on something in the basis set. Back, when ibm's basis was u1, u2, u3, cx this worked fine for ibm backends because every standard gate's definition was in terms of that basis. But it never worked for other backends or ibm's new basis of sx, x, rz, cx. This is why the unroller has been replaced by the basis translator (which uses a library of equivalence rules to translate between arbitrary basis sets) as the default everywhere.